streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorStateManager.java

```diff
@@ -415,6 +415,8 @@ public void close() throws ProcessorStateException {
                     log.error("Failed to close state store {}: ", store.name(), exception);
                 }
             }
+
+            stores.clear();
```

Comment:
We need to clear the stores map now since we may re-initialize the state stores upon reviving a task.

Reply:
Do we also need to clear `storeToChangelogTopic`, etc?

Reply:
`storeToChangelogTopic` and `sourcePartitions` are passed in at construction time and final, so we cannot clear them (since they would only be initialized once).